src/adaptors/adaptor_common.c

```diff
@@ -23,150 +23,68 @@
-            break;
-        }
-
-        if (config_ssl_profile->ssl_trusted_certificate_db) {
-            res = pn_tls_config_set_trusted_certs(*tls_config, config_ssl_profile->ssl_trusted_certificate_db);
-            if (res != 0) {
-                qd_log(log_source, QD_LOG_ERROR, "[C%"PRIu64"] %s %s unable to set tls trusted certificates (%d)", conn_id, role, config->name, res);
-                break;
-            }
-        }
-
-        // Call pn_tls_config_set_credentials only if "certFile" is provided.
-        if (config_ssl_profile->ssl_certificate_file) {
-            res = pn_tls_config_set_credentials(*tls_config,
-                                                config_ssl_profile->ssl_certificate_file,
-                                                config_ssl_profile->ssl_private_key_file,
-                                                config_ssl_profile->ssl_password);
-            if (res != 0) {
-                qd_log(log_source, QD_LOG_ERROR, "[C%"PRIu64"] HTTP2 %s %s unable to set tls credentials (%d)", conn_id, role, config->name, res);
-                break;
-            }
-        }
-        else {
-            qd_log(log_source, QD_LOG_INFO, "[C%"PRIu64"] sslProfile %s did not provide certFile", conn_id, config->ssl_profile_name);
-        }
+//size_t QD_ADAPTOR_MAX_BUFFER_SIZE = 4096;
+//size_t QD_ADAPTOR_MAX_BUFFER_SIZE = 8192;
+//size_t QD_ADAPTOR_MAX_BUFFER_SIZE = 16384;
+size_t QD_ADAPTOR_MAX_BUFFER_SIZE = 32768;
+//size_t QD_ADAPTOR_MAX_BUFFER_SIZE = 65536;
 
 
-        if (!!config_ssl_profile->ssl_ciphers) {
-            res = pn_tls_config_set_impl_ciphers(*tls_config, config_ssl_profile->ssl_ciphers);
-            if (res != 0) {
-                qd_log(log_source, QD_LOG_ERROR, "[C%"PRIu64"] %s %s unable to set tls ciphers (%d)", conn_id, role, config->name, res);
-                break;
-            }
-        }
-
-
-        if (is_listener) {
-            if (config->authenticate_peer) {
-                res = pn_tls_config_set_peer_authentication(*tls_config, PN_TLS_VERIFY_PEER, config_ssl_profile->ssl_trusted_certificate_db);
-            }
-            else {
-                res = pn_tls_config_set_peer_authentication(*tls_config, PN_TLS_ANONYMOUS_PEER, 0);
-            }
-        }
-        else {
-            // Connector.
-            if (config->verify_host_name) {
-                res = pn_tls_config_set_peer_authentication(*tls_config, PN_TLS_VERIFY_PEER_NAME, config_ssl_profile->ssl_trusted_certificate_db);
-            }
-            else {
-                res = pn_tls_config_set_peer_authentication(*tls_config, PN_TLS_VERIFY_PEER, config_ssl_profile->ssl_trusted_certificate_db);
-            }
-
-            *tls_has_output = true; // always true for initial client side TLS.
-        }
+qd_adaptor_buffer_t *qd_adaptor_buffer()
+{
+    qd_adaptor_buffer_t *adaptor_buff = new_qd_adaptor_buffer_t();
+    ZERO(adaptor_buff);
```

Comment:
Probably best to not zero the entire buffer, just initialize the structure fields instead.  Zeroing a 38K buffer on every allocation will be expensive.